lld/ELF/Relocations.cpp

```diff
@@ -1352,6 +1352,36 @@ unsigned RelocationScanner::handleTlsRelocation(RelExpr expr, RelType type,
     return 1;
   }
 
+  auto fatalBothAuthAndNonAuth = [&sym]() {
+    fatal("both AUTH and non-AUTH TLSDESC entries for '" + sym.getName() +
```

Comment:
Can you add getLocation to the error message so that a user can find the source of at least one of the relocations?

I also recommend not using fatal but use error so that a user can get diagnostics like the map file out of the link.

Reply:
We now use `Fatal(ctx) << ...`. Unlike fatal, `Fatal` executes `exit` but is not `noreturn`.

`Fatal` should generally be avoided in favor of `Err`.

Reply:
> We now use `Fatal(ctx) << ...`. Unlike fatal, `Fatal` executes `exit` but is not `noreturn`.
> 
> `Fatal` should generally be avoided in favor of `Err`. You will need to make sure lld doesn't crash when it keeps execution, though the output is not required to be functional.

Thanks! Switched from `fatal` to `Err(ctx)`. After emitting an error, we return the number of processed relocations (which is 1), and do not add any entries to GOT synthetic section. So, I don't expect lld to crash, and it should be OK to use `Err` instead of `Fatal` here. See 105213dde62927e619fb38149e8c613ab7462ed8

> Can you add getLocation to the error message so that a user can find the source of at least one of the relocations?

Added that, thanks! See 105213dde62927e619fb38149e8c613ab7462ed8